### PR TITLE
Merges WPAndroid changes to the login library from base repository

### PR DESCRIPTION
**Pull request by oguzkocer**
wordpress-mobile/WordPress-Login-Flow-Android#57 merges changes from both WPAndroid and WCAndroid for login library. This PR brings in those changes so both clients and the base repo is in the same state.

I'll add line comments for where each change has been made in WPAndroid. Note that, this does NOT mean that every WPAndroid change is the right one for WCAndroid. Part of this PR review/testing process is to figure out if there has been any changes to the library that's not valid for either client. Please make sure to pay extra attention to this review.

**To test:**
* This was a very error prone merge, so I think we should test the whole login flow

**Update release notes:**

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: libs/login/WordPressLoginFlow/src/main/res/layout/toolbar_login.xml [hunk 1]
```diff
@@ -17,9 +17,9 @@
             android:layout_height="24dp"
             android:layout_gravity="center"
             android:contentDescription="@null"
-            android:tint="?attr/colorControlNormal"
             android:visibility="gone"
-            app:srcCompat="@drawable/login_toolbar_icon" />
+            app:srcCompat="@drawable/login_toolbar_icon"
+            app:tint="?attr/colorControlNormal" />
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/res/layout/toolbar_login.xml:22]: This is just a change from `android:tint` to `app:tint` I've manually made to fix the lint error in the library.

File: libs/login/gradlew.bat [hunk 2]
```diff
@@ -1,103 +1,100 @@
-@rem
```
Comment (oguzkocer) [inline libs/login/gradlew.bat:1]: The changes in this file have been made by running `./gradlew wrapper --gradle-version 6.1.1 --distribution-type all` and should be ignored.

File: libs/login/WordPressLoginFlow/build.gradle [hunk 3]
```diff
@@ -65,7 +65,7 @@ dependencies {
             exclude group: "org.wordpress", module: "utils"
         }
     } else {
-        implementation("com.github.wordpress-mobile.WordPress-FluxC-Android:fluxc:1.6.22") {
+        implementation("com.github.wordpress-mobile.WordPress-FluxC-Android:fluxc:1.17.0") {
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/build.gradle:68]: I've manually updated the default FluxC version to the latest version which fixes the lazysodium dependency issue. (Bintray shutting down and lazysodium moving to mavenCentral)

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseFormFragment.java [hunk 4]
```diff
@@ -40,6 +40,7 @@
 import org.wordpress.android.fluxc.store.SiteStore.FetchSitesPayload;
 import org.wordpress.android.fluxc.store.SiteStore.OnSiteChanged;
 import org.wordpress.android.fluxc.store.SiteStore.SiteErrorType;
+import org.wordpress.android.login.util.SiteUtils;
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginBaseFormFragment.java:43]: Changes in this file has been made in this WPAndroid PR: https://github.com/wordpress-mobile/WordPress-Android/pull/14356/files#diff-b4d2aa999e9c726520bbc44a04255321d23afc5509e131926d32f44764297cfc. `isJetpackAppLogin` has later been marked as `protected` in this PR: https://github.com/wordpress-mobile/WordPress-Android/pull/14662/files#diff-b4d2aa999e9c726520bbc44a04255321d23afc5509e131926d32f44764297cfc

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginEmailFragment.java [hunk 5]
```diff
@@ -156,6 +156,7 @@ protected void setupLabel(@NonNull TextView label) {
                 break;
             case FULL:
             case WPCOM_LOGIN_ONLY:
+            case JETPACK_LOGIN_ONLY:
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginEmailFragment.java:159]: Added here: https://github.com/wordpress-mobile/WordPress-Android/pull/14459/files#diff-62e7b70e948b55d43cb3e64254f0b7bb8fb4a0afe86e585f1e30bc6d3536581e

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginEmailPasswordFragment.java [hunk 6]
```diff
@@ -278,8 +278,15 @@ protected void next() {
...
-                                                    mIsSocialLogin);
+        LoginWpcomService.loginWithEmailAndPassword(
+                getContext(),
+                mEmailAddress,
+                mRequestedPassword,
+                mIdToken,
+                mService,
+                mIsSocialLogin,
+                mLoginListener.getLoginMode() == LoginMode.JETPACK_LOGIN_ONLY
+        );
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginEmailPasswordFragment.java:289]: https://github.com/wordpress-mobile/WordPress-Android/pull/14356/files#diff-3f5af36ec5ca9aa8562eaccabb665e45737564ce5d8eb22f2794931fa8175e92

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginMode.java [hunk 7]
```diff
@@ -6,6 +6,7 @@
     FULL,
     SELFHOSTED_ONLY,
     WPCOM_LOGIN_ONLY,
+    JETPACK_LOGIN_ONLY,
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginMode.java:9]: https://github.com/wordpress-mobile/WordPress-Android/pull/14356/files#diff-a2033f3437b504685079d706b43b163ff3f22a1a0de604291929bb48ca1e2c07

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginSiteAddressFragment.java [hunk 8]
```diff
@@ -86,13 +86,10 @@
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginSiteAddressFragment.java:86]: Changes in this file has been made here: https://github.com/wordpress-mobile/WordPress-Android/pull/14459/files#diff-9310487644e82961ebeb2cb61c2f3d4e4fa510c92799fdbd7f75654eaf80bad3

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginUsernamePasswordFragment.java [hunk 9]
```diff
@@ -505,7 +508,7 @@ private void handleAuthError(AuthenticationErrorType error, String errorMessage)
                 AppLog.e(T.NUX, "Server response: " + errorMessage);
 
                 ToastUtils.showToast(getActivity(),
-                        errorMessage == null ? getString(R.string.error_generic) : errorMessage);
+                        TextUtils.isEmpty(errorMessage) ? getString(R.string.error_generic) : errorMessage);
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginUsernamePasswordFragment.java:511]: https://github.com/wordpress-mobile/WordPress-Android/pull/14630/files

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginUsernamePasswordFragment.java [hunk 10]
```diff
@@ -117,8 +117,10 @@ public static LoginUsernamePasswordFragment newInstance(String inputSiteAddress,
     protected void setupLabel(@NonNull TextView label) {
         final boolean isWoo = mLoginListener.getLoginMode() == LoginMode.WOO_LOGIN_MODE;
         final int labelResId = isWoo ? R.string.enter_credentials_for_site : R.string.enter_account_info_for_site;
+        final String siteAddress =
+                (mEndpointAddress == null || mEndpointAddress.isEmpty()) ? mInputSiteAddress : mEndpointAddress;
         final String formattedSiteAddress =
-                UrlUtils.removeScheme(UrlUtils.removeXmlrpcSuffix(StringUtils.notNullStr(mInputSiteAddress)));
+                UrlUtils.removeScheme(UrlUtils.removeXmlrpcSuffix(StringUtils.notNullStr(siteAddress)));
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginUsernamePasswordFragment.java:123]: https://github.com/wordpress-mobile/WordPress-Android/pull/14486/files

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginWpcomService.java [hunk 11]
```diff
@@ -26,6 +26,7 @@
 import org.wordpress.android.fluxc.store.SiteStore.OnSiteChanged;
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/LoginWpcomService.java:26]: Changes in this file has been made in https://github.com/wordpress-mobile/WordPress-Android/pull/14356/files#diff-d0c8aefa7f701c0e37f0d6cfc4eb281a2285f7e6282656792f19e890109e21b1.

File: libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/util/SiteUtils.java [hunk 12]
```diff
@@ -1,9 +1,12 @@
 package org.wordpress.android.login.util;
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/java/org/wordpress/android/login/util/SiteUtils.java:1]: Changes in this file has been made in https://github.com/wordpress-mobile/WordPress-Android/pull/14356/files#diff-a1c243e1fdf3aeabdb1a3b5ddbf83bdf669c9e0f9ddd6c28a948e8e96cb2e737.

File: libs/login/WordPressLoginFlow/src/main/res/drawable/login_site_address_help.xml [hunk 13]
```diff
@@ -1,39 +1,113 @@
-<vector android:height="24dp" android:viewportHeight="262.0"
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/res/drawable/login_site_address_help.xml:1]: https://github.com/wordpress-mobile/WordPress-Android/pull/14643

File: libs/login/WordPressLoginFlow/src/main/res/values-night/themes.xml [hunk 14]
```diff
@@ -2,10 +2,10 @@
...
 
     <style name="Base.Theme.LoginFlow" parent="Base.Theme.LoginFlow.Shared">
-        <item name="colorPrimary">@color/wp_blue_40</item>
-        <item name="colorPrimaryVariant">@color/wp_blue_70</item>
-        <item name="colorSecondary">@color/wp_pink_50</item>
-        <item name="colorSecondaryVariant">@color/wp_pink_70</item>
+        <item name="colorPrimary">@color/wp_blue_30</item>
+        <item name="colorPrimaryVariant">@color/wp_blue_50</item>
+        <item name="colorSecondary">@color/wp_blue_30</item>
+        <item name="colorSecondaryVariant">@color/wp_blue_50</item>
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/res/values-night/themes.xml:8]: https://github.com/wordpress-mobile/WordPress-Android/pull/14410/files

File: libs/login/WordPressLoginFlow/src/main/res/values/colors.xml [hunk 15]
```diff
@@ -1,9 +1,9 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
     <!-- Theme Colors -->
-    <color name="wp_blue_40">#3582c4</color>
-    <color name="wp_blue_50">#2271b1</color>
-    <color name="wp_blue_70">#0a4b78</color>
+    <color name="wp_blue_30">#399ce3</color>
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/res/values/colors.xml:4]: https://github.com/wordpress-mobile/WordPress-Android/pull/14410/files

File: libs/login/WordPressLoginFlow/src/main/res/values/colors.xml [hunk 16]
```diff
@@ -1,9 +1,9 @@
...
+    <color name="wp_blue_50">#0675c4</color>
+    <color name="wp_blue_70">#044b7a</color>
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/res/values/colors.xml:6]: https://github.com/wordpress-mobile/WordPress-Android/commit/12dbb8666b9ac97dc294e38ca1f853fc307ab28a#diff-b3b5fed18330c04ab44e7dbef4297c4f3afb14a7534a6592af2b4fd7713e41fb

File: libs/login/WordPressLoginFlow/src/main/res/values/themes.xml [hunk 17]
```diff
@@ -47,8 +47,8 @@
     <style name="Base.Theme.LoginFlow" parent="Base.Theme.LoginFlow.Shared">
         <item name="colorPrimary">@color/wp_blue_50</item>
         <item name="colorPrimaryVariant">@color/wp_blue_70</item>
-        <item name="colorSecondary">@color/wp_pink_50</item>
-        <item name="colorSecondaryVariant">@color/wp_pink_70</item>
+        <item name="colorSecondary">@color/wp_blue_50</item>
+        <item name="colorSecondaryVariant">@color/wp_blue_70</item>
```
Comment (oguzkocer) [inline libs/login/WordPressLoginFlow/src/main/res/values/themes.xml:51]: https://github.com/wordpress-mobile/WordPress-Android/pull/14376/files#diff-16d81f626ddf61b558199c724ec9c8644cd8a42f2cc3be9936fbfc543c9443e4

**Review by anitaa1990 — APPROVED** [review]
Tested the following scenarios in WCAndroid:
✅ Login with `Continue with WordPress.com` -> Enter valid email address + password
✅ Login with `Continue with WordPress.com` -> Enter valid email address + magic link login 
✅ Login with `Enter your store address` -> Enter valid email address + password
✅ Login with `Enter your store address` -> Enter valid email address + magic link login
✅ Login with `Enter your store address` -> Enter a url that does not have WordPress -> Error screen is displayed.
✅ Login with `Enter your store address` -> Enter a url with no Jetpack installed -> Correct error screen is displayed.
✅ Login with `Enter your store address` -> Enter an invalid email address -> correct error message is displayed
✅ Login with `Enter your store address` -> Continue with store credentials -> enter valid username + password
✅ Login with `Enter your store address` -> Continue with store credentials -> enter invalid username or password -> Correct error message is displayed.
✅ Reset password works as expected.
✅ Entered a non Woo site and error message is displayed.

All the help buttons and text are displayed as expected. The code changes look good 👍

**Review by malinajirka — APPROVED** [review]
I've reviewed the changes and tested a few different login flows and everything worked as expected. Great job ;)!

**Pull request merged**